common/common/relations/database_provides.py

```diff
@@ -30,12 +30,25 @@ class _UnsupportedExtraUserRole(status_exception.StatusException):
 
     def __init__(self, *, app_name: str, endpoint_name: str) -> None:
         message = (
-            f"{app_name} app requested unsupported extra user role on {endpoint_name} endpoint"
+            f"{app_name} app requested unsupported extra user role on "
+            f"{endpoint_name} endpoint"
         )
         logger.warning(message)
         super().__init__(ops.BlockedStatus(message))
 
 
+class _InvalidDatabaseName(status_exception.StatusException):
+    """Application charm requested an invalid database name"""
+
+    def __init__(self, *, app_name: str, endpoint_name: str, exception_msg: str) -> None:
+        message = (
+            f"{app_name} app requested an invalid database name on "
+            f"{endpoint_name} endpoint: {exception_msg}"
```

Comment:
```suggestion
            f"{app_name} app requested invalid database name >20 characters on "
            f"{endpoint_name} endpoint"
```
to keep message short (under 120 characters) so it doesn't get truncated in `juju status` & to give user clear actionable next step (request a shorter database name)

Reply:
I would rather have a distinction between the log message (long format), and the message displayed when doing `juju status` (short format). As it is proposed at the moment:

- Log: _{app} requested an invalid database name on {endpoint} endpoint: DBA role longer than 32 characters_
- Status: _DBA role longer than 32 characters_.

Reply:
That's good alternative.